From 0c452fee7e3abb8e117bd390610f39b95069dc1b Mon Sep 17 00:00:00 2001
From: Andrei Vagin <avagin@google.com>
Date: Tue, 14 Dec 2021 08:19:16 -0800
Subject: [PATCH 071/245] usernsd: UNS_FDOUT should not require an input
 descriptor

UNS_FDOUT means only that a userns call will return a file descriptor.

Signed-off-by: Andrei Vagin <avagin@google.com>
---
 criu/namespaces.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/criu/namespaces.c b/criu/namespaces.c
index 7fa58682b..c36e631cf 100644
--- a/criu/namespaces.c
+++ b/criu/namespaces.c
@@ -1330,11 +1330,6 @@ static int usernsd(int sk)
 		unsc_msg_pid_fd(&um, &pid, &fd);
 		pr_debug("uns: daemon calls %p (%d, %d, %x)\n", call, pid, fd, flags);
 
-		if (fd < 0 && flags & UNS_FDOUT) {
-			pr_err("uns: bad flags/fd %p %d %x\n", call, fd, flags);
-			BUG();
-		}
-
 		/*
 		 * Caller has sent us bare address of the routine it
 		 * wants to call. Since the caller is fork()-ed from the
-- 
2.35.1

